File: Chapter 03/C#/3.3.1 SRP/Before/Player.cs
using System;

namespace Chapter03.SRP.Before;

public class Player
{
    public required string Name { get; set; }

    // Responsibility 1: Player’s own state/abilities
    public void DribbleBall() 
    {
        Console.WriteLine($"  [Action] {Name} is dribbling the ball down the court.");
    }

    // Responsibility 2: Tactical Logic
    public void DetermineBestPosition() 
    {
        Console.WriteLine($"  [Tactics] Calculating optimal court position for {Name}...");
    }

    // Responsibility 3: Data Persistence
    public void SaveStatsToDatabase()
    {
        Console.WriteLine($"  [Database] Saving {Name}'s game stats to the database.");
    }
}

File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/3.3.1 SRP/Before/Player.java
package com.grokkingsoftwarearchitecture.chapter03.srp.before;

public class Player {
    public String name;

    public Player(String name) {
        this.name = name;
    }

    // Responsibility 1: Player’s own state/abilities
    public void dribbleBall() {
        System.out.println("  [Action] " + name + " is dribbling the ball down the court.");
    }

    // Responsibility 2: Tactical Logic
    public void determineBestPosition() {
        System.out.println("  [Tactics] Calculating optimal court position for " + name + "...");
    }

    // Responsibility 3: Data Persistence
    public void saveStatsToDatabase() {
        System.out.println("  [Database] Saving " + name + "'s game stats to the database.");
    }
}